drivers/serial/uart_nrf_sw_lpuart.c

```diff
@@ -84,10 +63,12 @@ struct lpuart_data {
 	const struct device *uart;
 
 	/* Request pin. */
-	struct lpuart_bidir_gpio req_pin;
-
+	nrfx_gpiote_pin_t req_pin;
 	/* Response pin. */
-	struct lpuart_bidir_gpio rdy_pin;
+	nrfx_gpiote_pin_t rdy_pin;
```

Comment:
Why are these pin numbers stored here? Couldn't the fields from `struct lpuart_config` be used instead?

Reply:
i started to convert that because it seemed like you were right but there is this one case where uart is driven from onoff clock control callback. In that case by using `CONTAINER_OF` i can get to data structure but have no access to config structure. I would need to add pointer to config structure to data to have access to rdy pin. So for req pin it is possible but not for rdy pin. So imo it's better to leave it as is and have both pins in data.